File: 2164. Sort Even and Odd Indices Independently/solution.java
class Solution {
  public int[] sortEvenOdd(int[] nums) {
    List<Integer> oddIndex = new ArrayList<Integer>();
    List<Integer> evenIndex = new ArrayList<Integer>();
    
    for (int index = 0; index < nums.length; index++){
      int num = nums[index];
      if (index % 2 == 0){
        evenIndex.add(num);
      }
      else{
        oddIndex.add(num);
      }
    }
    
    Collections.sort(oddIndex);
    Collections.sort(evenIndex);

    int[] result = new int[nums.length];
    int count = 0;
    for (int index = 0; index < oddIndex.size(); index++){
      int even = evenIndex.get(index);
      int odd = oddIndex.get(oddIndex.size() - 1 - index);
      result[count++] = even;
      result[count++] = odd;
    }

    if (nums.length % 2 != 0){
      result[count++] = evenIndex.get(evenIndex.size() - 1);
    }

    return result;
  }
}

File: 2164. Sort Even and Odd Indices Independently/solution.cpp
class Solution {
public:
  vector<int> sortEvenOdd(vector<int>& nums) {
    vector<int> oddIndex;
    vector<int> evenIndex;
    for (int index = 0; index < nums.size(); index++){
      int num = nums[index];
      if (index % 2 == 0){
        evenIndex.push_back(num);
      }
      else{
        oddIndex.push_back(num);
      }
    }
    
    sort(oddIndex.begin(), oddIndex.end());
    sort(evenIndex.begin(), evenIndex.end());

    vector<int> result;
    for (int index = 0; index < oddIndex.size(); index++){
      int even = evenIndex[index];
      int odd = oddIndex[oddIndex.size() - 1 - index];
      result.push_back(even);
      result.push_back(odd);
    }

    if (nums.size() % 2 != 0){
      result.push_back(evenIndex[evenIndex.size() - 1]);
    }

    return result;
  }
};